src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java

```diff
@@ -0,0 +1,117 @@
+     * The exception is thrown anyway if 10 times retry did not succeed.
+     *
+     * @param config The PermissionConfig to be used
+     * @param expectedStatus list of allowed HTTP Status to be returned. If not set,
+     *                       http status 200 (OK) is assumed.
+     * @return Json node containing resulting permissions
+     * @throws ClientException
+     *          If something fails during request/response cycle after 10 times unsuccessful retry
+     */
+    public JsonNode changePermissionsWithRetry(PermissionConfig config, long timeout, long delay,
+                                               int... expectedStatus) throws ClientException, InterruptedException {
+
+        class ChangePermissionsPolling extends Polling {
+            JsonNode permissions = JsonNodeFactory.instance.objectNode();
+
+            @Override
+            public Boolean call() throws Exception {
+                try {
+                    permissions = changePermissions(config.getAuthorizableId(), config.getPath(),
+                            config.isRead(), config.isModify(), config.isCreate(), config.isDelete(),
+                            config.isAclRead(), config.isAclEdit(), config.isReplicate(), expectedStatus);
+                    LOG.info("new set permission response string for {} at {}: {}",
+                            config.getAuthorizableId(), config.getPath(), permissions);
+                    return true;
+                } catch (Exception e) {
+                        LOG.warn("Error on change permission -> retry:  {}", e.getLocalizedMessage(), e);
+                        throw e;
+                }
+            }
+        }
+
+        ChangePermissionsPolling createPolling = new ChangePermissionsPolling();
+        try {
+            createPolling.poll(timeout, delay);
+        } catch(TimeoutException e) {
+            String errorMsg = String.format("Failed to change permission for {} at {} in {}",
+                    config.getAuthorizableId(), config.getPath(), createPolling.getWaited());
+            throw new ClientException(errorMsg, e);
+        }
+        return createPolling.permissions;
+    }
+
+    /**
+     * Get permissions for an authorizable with retry in case an exception is thrown.
+     * The exception is thrown anyway if 10 times retry did not succeed.
+     *
+     * @param authorizableId the Id of the authorizable
+     * @param path           path
+     * @param depth          depth
+     * @param expectedStatus list of allowed HTTP Status to be returned. If not set,
+     *                       http status 200 (OK) is assumed.
+     * @return the root {@link JsonNode}
+     * @throws ClientException If something fails during request/response cycle after 10 times unsuccessful retry
+     */
+    @Override
+    public JsonNode getPermissions(String authorizableId, String path, int depth, int... expectedStatus) throws
```

Comment:
Same polling mechanism here as for create, please. Sorry that i missed this in the previous review.